xclim/indices/generic.py

```diff
@@ -147,97 +147,128 @@ def fitfunc(arr):
+    out.attrs["original_name"] = da.attrs.get("standard_name", "")
+    out.attrs["original_units"] = da.attrs.get("units", "")
     out.attrs[
         "description"
-    ] = f"Parameters of the {dist} distribution fitted over {getattr(da, 'standard_name', '')}"
+    ] = f"Parameters of the {dist} distribution fitted over {out.attrs['original_name']}"
     out.attrs["estimator"] = "Maximum likelihood"
     out.attrs["scipy_dist"] = dist
     out.attrs["units"] = ""
-    # out.name = 'params'
+
     return out
 
 
-def fa(
-    da: xr.DataArray, t: Union[int, Sequence], dist: str = "norm", mode: str = "high"
-):
-    """Return the value corresponding to the given return period.
+def parametric_quantile(p: xr.DataArray, q: Union[int, Sequence]):
+    """Return the value corresponding to the given distribution parameters and quantile.
 
     Parameters
     ----------
-    da : xr.DataArray
-      Maximized/minimized input data with a `time` dimension.
-    t : Union[int, Sequence]
-      Return period. The period depends on the resolution of the input data. If the input array's resolution is
-      yearly, then the return period is in years.
-    dist : str
-      Name of the univariate distribution, such as beta, expon, genextreme, gamma, gumbel_r, lognorm, norm
-      (see scipy.stats).
-    mode : {'min', 'max}
-      Whether we are looking for a probability of exceedance (max) or a probability of non-exceedance (min).
+    p : xr.DataArray
+      Distribution parameters returned by the `fit` function. The array should have dimension `dparams` storing the
+      distribution parameters, and attribute `scipy_dist`, storing the name of the distribution.
+    q : Union[float, Sequence]
+      Quantile to compute, which must be between 0 and 1 inclusive.
 
     Returns
     -------
     xarray.DataArray
-      An array of values with a 1/t probability of exceedance (if mode=='max').
+      An array of parametric quantiles estimated from the distribution parameters.
+
+    Notes
+    -----
+    When all quantiles are above 0.5, the `isf` method is used instead of `ppf` because accuracy is sometimes better.
     """
-    t = np.atleast_1d(t)
+    q = np.atleast_1d(q)
 
     # Get the distribution
+    dist = p.attrs["scipy_dist"]
     dc = get_dist(dist)
 
-    # Fit the parameters of the distribution
-    p = fit(da, dist)
-
     # Create a lambda function to facilitate passing arguments to dask. There is probably a better way to do this.
-    if mode in ["max", "high"]:
+    if np.all(q > 0.5):
 
         def func(x):
-            return dc.isf(1.0 / t, *x)
+            return dc.isf(1 - q, *x)
 
-    elif mode in ["min", "low"]:
+    else:
 
         def func(x):
-            return dc.ppf(1.0 / t, *x)
-
-    else:
-        raise ValueError(f"Mode `{mode}` should be either 'max' or 'min'.")
+            return dc.ppf(q, *x)
 
     data = dask.array.apply_along_axis(func, p.get_axis_num("dparams"), p)
```

Comment:
Not from your changes, but we could use the duck typing approach here to I think. So that the output is not always a dask array.

Reply:
Done in `fix-33`.

Reply:
Oups, no, I did it for fit and fit_pwm, not for parametric_quantile.... will do.